### PR TITLE
Add inference colorization

### PR DESCRIPTION
**Pull request by klei22**
This adds two colorization modes for sample.py:

1. 'minmax' - direct logit magnitude, but normalized over inference.
2. 'softmax" - normalized over all vocab logits with softmax, shows even lower values for first letter.
3. 'softmax_top_k' - normalized first by softmax then printing colors, at normal temperatures of softmax it shows moreso non-top-k choices.

Left minmax, right softmax 
image

Left softmax_top_k right softmax:
image

### CODE REVIEW

**Review by gkielian — APPROVED** [review]
Looks good
